### PR TITLE
Manik tharaka add plotly capabilities

### PR DESCRIPTION
**Pull request by manikTharaka**
Added tests and plot parameters functionality

### CODE REVIEW

File: neuralprophet/plot_model_parameters_plotly.py [hunk 1]
```diff
@@ -0,0 +1,669 @@
...
+    else:
+        if len(weights.shape) == 2:
+            weights = weights[focus - 1, :]
+
+        traces.append(go.Bar(name=comp_name, x=lags_range, y=weights, marker_color=color, width=0.8))
+
+    xaxis = go.layout.XAxis(title=f"{comp_name} lag number")
+
+    if focus is None:
+        ax.set_ylabel("{} relevance".format(comp_name))
```
Comment (lsy44415) [inline neuralprophet/plot_model_parameters_plotly.py:361]:
```suggestion
        
```
Delete ax.set_ylabel(), not used in plotly

File: neuralprophet/plot_model_parameters_plotly.py [hunk 2]
```diff
@@ -0,0 +1,669 @@
...
+    else:
+        predicted = m.predict_seasonal_components(df_w)[comp_name]
+    days = pd.date_range(start="2017-01-01", periods=7) + pd.Timedelta(days=weekly_start)
+    days = days.day_name()
+
+    traces.append(
+        go.Scatter(
+            name=comp_name,
+            x=range(len(days_i), y=predicted, mode="lines", line=dict(color=color, width=line_width), fill="none"),
+        )
```
Comment (lsy44415) [inline neuralprophet/plot_model_parameters_plotly.py:467]:
```suggestion
           x=list(range(len(days_i))), y=predicted, mode="lines", line=dict(color=color, width=line_width), fill="none"),
```
x cannot be the range type, should be a list

File: neuralprophet/plot_model_parameters_plotly.py [hunk 3]
```diff
@@ -0,0 +1,669 @@
...
+    traces.append(
+        go.Scatter(
+            name=comp_name,
+            x=range(
+                df_y["ds"].dt.to_pydatetime(),
+                y=predicted,
+                mode="lines",
+                line=dict(color=color, width=line_width),
+                fill="none",
+            ),
```
Comment (lsy44415) [inline neuralprophet/plot_model_parameters_plotly.py:415]:
```suggestion
            x=df_y["ds"].dt.to_pydatetime(),
            y=predicted,
            mode="lines",
            line=dict(color=color, width=line_width),
            fill="none",
```
Remove range(), datetime cannot be converted to a scalar index

File: neuralprophet/plot_model_parameters_plotly.py [hunk 4]
```diff
@@ -0,0 +1,669 @@
...
+    line_width = 1
+    zeroline_color = "#AAA"
+
+    t_i, predicted = predict_one_season(m, name=comp_name, n_steps=300)
+    traces = []
+
+    traces.append(
+        go.Scatter(
+            name=comp_name,
+            x=range(t_i, y=predicted, mode="lines", line=dict(color=color, width=line_width), fill="none"),
```
Comment (lsy44415) [inline neuralprophet/plot_model_parameters_plotly.py:555]:
```suggestion
            x=range(t_i), y=predicted, mode="lines", line=dict(color=color, width=line_width), fill="none",
```
Miss right bracket of range

File: neuralprophet/plot_model_parameters_plotly.py [hunk 5]
```diff
@@ -0,0 +1,669 @@
...
+        predicted = predict_season_from_dates(m, dates=df["ds"], name=comp_name)
+    else:
+        predicted = m.predict_seasonal_components(df)[comp_name]
+
+    traces.append(
+        go.Scatter(
+            name=comp_name,
+            x=range(
+                range(len(dates)), y=predicted, mode="lines", line=dict(color=color, width=line_width), fill="none"
+            ),
```
Comment (lsy44415) [inline neuralprophet/plot_model_parameters_plotly.py:516]:
```suggestion
            x=range(len(dates)), y=predicted, mode="lines", line=dict(color=color, width=line_width), fill="none"
```
Duplicate range()

File: neuralprophet/plot_model_parameters_plotly.py [hunk 6]
```diff
@@ -0,0 +1,669 @@
...
+
+    Returns:
+        A plotly figure.
+    """
+
+    parameter_components = get_parameter_components(m, forecast_in_focus)
+
+    components = parameter_components["components"]
+    additive_future_regressors = parameter_components["additive_future_regressors"]
+    additive_events = parameter_component["additive_events"]
```
Comment (lsy44415) [inline neuralprophet/plot_model_parameters_plotly.py:595]:
```suggestion
    additive_events = parameter_components["additive_events"]
```
missing s

File: neuralprophet/plot_model_parameters_plotly.py [hunk 7]
```diff
@@ -0,0 +1,669 @@
...
+                trace_object = plot_yearly(
+                    m=m, yearly_start=yearly_start, comp_name=name, multiplicative=is_multiplicative
+                )
+            elif name.lower() == "daily" or m.season_config.periods[name].period == 1:
+                trace_object = plot_daily(m=m, comp_name=name, multiplicative=is_multiplicative)
+            else:
+                trace_object = plot_custom_season(m=m, ax=ax, comp_name=name, multiplicative=is_multiplicative)
+        elif plot_name == "lagged weights":
+            trace_object = plot_lagged_weights(
+                weights=comp["weights"], comp_name=comp["comp_name"], focus=comp["focus"], ax=ax
```
Comment (lsy44415) [inline neuralprophet/plot_model_parameters_plotly.py:638]:
```suggestion
                weights=comp["weights"], comp_name=comp["comp_name"], focus=comp["focus"]
```
Remove unnecessary ax=ax

File: neuralprophet/plot_plotly.py [hunk 8]
```diff
@@ -0,0 +1,646 @@
...
+                print(comp_name)
+                trace_object = get_seasonality_plotly_props(m, fcst, **comp)
+            else:
+                comp_name = f"season_{comp['comp_name']}"
+                trace_object = get_forecast_component_plotly_props(
+                    m, fcst=fcst, comp_name=comp_name, plot_name=comp["plot_name"]
+                )
+
+        elif "auto-regression" in name or "lagged regressor" in name or "residuals" in name:
+            trace_object = get_multiforecast_component_plotly_props(fcst=fcst, **comp)
```
Comment (lsy44415) [inline neuralprophet/plot_plotly.py:314]:
```suggestion
            trace_object = get_multiforecast_component_plotly_props(fcst=fcst, **comp)
            fig.update_layout(barmode='overlay')
```
Set the barchart mode as overlay

File: neuralprophet/plot_plotly.py [hunk 9]
```diff
@@ -0,0 +1,646 @@
...
+            else:
+                y[-1] = 0
+
+            if bar:
+                traces.append(
+                    go.Bar(
+                        name=plot_name,
+                        x=fcst_t,
+                        y=y,
+                        width=[1 for i in fcst_t],
```
Comment (lsy44415) [inline neuralprophet/plot_plotly.py:506]:
```suggestion
                        width=[1 for i in fcst_t],
```
```suggestion
                        #width=[1 for i in fcst_t],
```

Comment (lsy44415) [inline neuralprophet/plot_plotly.py:506]:
```suggestion
                        #width=[1 for i in fcst_t],
```

File: neuralprophet/plot_plotly.py [hunk 10]
```diff
@@ -0,0 +1,646 @@
...
+        else:
+            y[-1] = 0
+        if bar:
+            traces.append(
+                go.Bar(
+                    name=plot_name,
+                    x=fcst_t,
+                    y=y,
+                    text=text,
+                    width=[1 for i in fcst_t],
```
Comment (lsy44415) [inline neuralprophet/plot_plotly.py:542]:
```suggestion
                    #width=[1 for i in fcst_t],
```

File: neuralprophet/plot_model_parameters_plotly.py (hunk 3 above)
Comment (manikTharaka) [reply neuralprophet/plot_model_parameters_plotly.py:415]: Fixed in a later commit

**Comment by manikTharaka** [comment]
@ourownstory  this looks good once some minor conflicts are resolved. Since this took quite a bit of time, not sure how much of it is inline with the current roadmap. Let me know if there are any changes that needs to be made.

**Comment by ourownstory** [comment]
@manikTharaka Sorry for taking so long to review this. It is still a relevant part of the evolution of the framework. I will try to find time to dig in soon.

**Comment by ourownstory** [comment]
@manikTharaka it is finally merged in #671!
Thank you so much for your hard work, it was much appreciated!!

**Pull request closed without merging**